src/domain/tradingview/TVDataProvider.ts

```diff
@@ -115,22 +120,33 @@ export class TVDataProvider {
     return bars.slice(bars.length - countBack, bars.length);
   }
 
-  async getBars(chainId: number, ticker: string, resolution: string, isStable: boolean, periodParams: PeriodParams) {
+  async getBars(
+    chainId: number,
+    ticker: string,
+    resolution: string,
+    isStable: boolean,
+    periodParams: PeriodParams
+  ): Promise<FromOldToNewArray<Bar>> {
     const period = this.supportedResolutions[resolution];
     const { from, to } = periodParams;
 
     try {
-      const bars = isStable
+      const bars: FromOldToNewArray<Bar> = isStable
         ? getStableCoinPrice(period, from, to)
         : await this.getTokenHistoryBars(chainId, ticker, period, periodParams);
 
-      return bars.map(formatTimeInBarToMs);
+      return bars.map(formatTimeInBarToMs) as FromOldToNewArray<Bar>;
```

Comment:
why do you need `as` here if `bars` already has this type?

Reply:
map removes the flavored type back to T[] instead of F<T[]>